src/test/java/io/cdap/plugin/ariba/source/AribaInputFormatTest.java

```diff
@@ -0,0 +1,243 @@
+  public void testCreateRecordReader() throws IOException, AribaException, InterruptedException {
+    AribaPluginConfig pluginConfig = new AribaPluginConfig("unit-test-ref-name",
+                                                           "https://openapi.au.cloud.ariba.com",
+                                                           "prod",
+                                                           "CloudsufiDSAPP-T",
+                                                           "SourcingProjectFactSystemView",
+                                                           "08ee0299-4849-42a4-8464-3abed75fc74e",
+                                                           "c3B5wvrEsjKucFGlGhKSWUDqDRGE2Wds",
+                                                           "xryi0757SU8pEyk7ePc7grc7vgDXdz8O",
+                                                           "2022-01-28T10:05:02Z", "2022-01-31T10:05:02Z");
+    aribaInputFormat = new AribaInputFormat();
+    ObjectMapper objectMapper = new ObjectMapper();
+    JsonNode crateJob = objectMapper.readTree(createJobResponse);
+    JsonNode jobData = objectMapper.readTree(createJobResponse1);
+    new Expectations(AribaServices.class) {
+      {
+        aribaServices.createJob(pluginConfig, anyString);
+        result = crateJob;
+        minTimes = 0;
+
+        aribaServices.fetchJobStatus(anyString, anyString);
+        result = jobData;
+        minTimes = 0;
+
+        aribaServices.getAccessToken();
+        result = "token";
+        minTimes = 0;
+      }
+    };
+    aribaInputFormat.createJob(pluginConfig, aribaServices, false, "kk");
+    Assert.assertNotNull(jobData);
+    Assert.assertEquals(jobData.size(), 4);
+  }
+
+  @Test
+  public void testCreateJobThrowError() throws IOException, AribaException, InterruptedException {
+    AribaPluginConfig pluginConfig = new AribaPluginConfig("unit-test-ref-name",
+                                                           "https://openapi.au.cloud.ariba.com",
+                                                           "prod",
+                                                           "CloudsufiDSAPP-T",
+                                                           "SourcingProjectFactSystemView",
+                                                           "08ee0299-4849-42a4-8464-3abed75fc74e",
+                                                           "c3B5wvrEsjKucFGlGhKSWUDqDRGE2Wds",
+                                                           "xryi0757SU8pEyk7ePc7grc7vgDXdz8O",
+                                                           "2022-01-28T10:05:02Z", "2022-01-31T10:05:02Z");
+    aribaInputFormat = new AribaInputFormat();
+    ObjectMapper objectMapper = new ObjectMapper();
+    JsonNode jobData = objectMapper.readTree(createJobResponse1);
+    new Expectations(AribaServices.class) {
+      {
+        aribaServices.createJob(pluginConfig, anyString);
+        result = new AribaException("limit exceed", 429);
+        minTimes = 0;
```

Comment:
what is minTimes?

Reply:
minTimes is the number of times this new Expectation block will run, since it starts count from zero here 0 means it will run 1 time.